src/rdkafka_assignor.c

```diff
@@ -208,7 +209,7 @@ int rd_kafka_assignor_topic_cmp (const void *_a, const void *_b) {
         const rd_kafka_assignor_topic_t *b =
                 *(const rd_kafka_assignor_topic_t * const *)_b;
 
-        return !strcmp(a->metadata->topic, b->metadata->topic);
+        return strcmp(a->metadata->topic, b->metadata->topic);
```

Comment:
Found this issue (which existed prior to this PR).

Reply:
LGTM. we need better testing!

Reply:
It was found by the tests added in this PR ;)